### ISSUE TITLE
Some specs

### ISSUE THREAD

**Issue by mcepl** [plan9-for-vimspace/plan9-for-vimspace#4]
I wrote to Paul:

> would it be possible to add somewhere to the repository some TODO or some description of what you would like to achieve? For the plan9 virgins (like me), it is hard to decipher what exactly is the idea.

and got this reply:

> Hi - best to raise an issue in the repo - that way, Felipe can also contribute.
>
> In the meantime, the basic idea is that we would like emulate a lot of what Russ Cox covers here:
>
> https://www.youtube.com/watch?v=dP1xVpMPn8M

I know the video, I had seen it couple of times already, however I am still confused.  I am not at all sure, how the concepts from ACME translate into the vim world. To start with the obvious, vim-universe is usually heavily keyboard-centric, whereas ACME is mouse-centric. I am afraid, that with [Zenclavier](http://archive.oreilly.com/pub/a/oreilly//news/zenclavier_1299.html) and all that, any attempt to make vim mouse-centric will meet with strong disfavor of the regular vim users (e.g., me ;)).

I guess, some kind of “this keyboard shortcut is an equivalent of this mouse chord” mapping will be necessary, right?

Or, other thought, do you plan to write some kind of ``plumber`` equivalent linking to the native MIME-differentiated applications?
Or do you want to use native vim equivalents (e.g., ``ftplugin/man.vim``)?

I think it would be very helpful to take the video, separate it into invidivual ideas, and then describe each in its original form, and how it should translate to the vim world. I see something of the kind of list in ``README``, but truth to be told, for the uninitiated it is completely incomprehensible, and even if I would have an idea, what you are talking about, I am not quite certain how to translate it to the vim-world.

**Comment by fmoralesc** [comment]
Well, I will need some time for giving a full reply, but just to begin:

1) "this keyboard shortcut is an equivalent of this mouse chord..." Yes, and this is already implemented. The system is a library of functions that is mapped to mouse actions and key mappings

2) "do you plan to write some kind of plumber equivalent". Yes, although no details are fixed yet.

In general, I think (and this is my personal view) what's interesting of acme is that it offers a very interesting mix of text editing and hypertext facilities: it makes text actionable, and this opens very interesting ways of working with it. In the vim world, porting acme to vim gives a different hypertext from that of plugins like vimwiki or some org-vim, where it is somehow more similar to hypertext in the web. 

The idea might still be somewhat cryptic, but I hope it is somewhat more clear what the point of it all is. I'll be back in a while.

**Comment by mcepl** [comment]
The working progress [paper](https://matej.ceplovi.cz/blog/drafts/design-notes-on-plan9-for-vimspace.html) on the topic is available. I will continue to work on it, time permitting, and perhaps one day move it out of the ``drafts`` status ;). Of course, comments are more than welcome, without them it doesn't make much sense to continue.

**Comment by fmoralesc** [comment]
Sure! I'm at a conference in Paris atmosphere, so I may be a little short
on time this week, but will get back to this ASAP. Thanks for putting down
your thoughts there.

**Comment by mcepl** [comment]
The first published version is on https://matej.ceplovi.cz/blog/design-notes-on-plan9-for-vimspace.html (or it will be, when I get the Apache back up and running ;)). Comments more than welcome (on the page, here, or to my email).

**Comment by mcepl** [comment]
Just to make a note here, https://github.com/aklt/rel.vim looks interesting.

**Comment by mcepl** [comment]
This looks interesting as well (for nvim only) https://github.com/fmoralesc/nvimfs : it looks like a way how to control *vim from the outside script.

**Comment by fmoralesc** [comment]
That looks familiar ;) Indeed, I created that as a prototype for things that I intended to add to this repo.

**Comment by mcepl** [comment]
> That looks familiar ;) Indeed, I created that as a prototype for things that I intended to add to this repo.

Yup, I have recognized the name only too late. Any hope it could be made working with vim8?

**Comment by fmoralesc** [comment]
It makes heavy use of neovim's remote plugin API, so I don't think so (the fuse filesystem has to be vim aware, and i don't know of any way to do that with vim8 without trying to run the filesystem within the embedded python interpreter, and I think that wouldn't work.) Honestly, I don't care much for vim8 compatibility myself, so I don't feel much pressure to working on that.

**Comment by mcepl** [comment]
> Honestly, I don't care much for vim8 compatibility myself, so I don't feel much pressure to working on that.

Sure, your decision, just curious.

**Comment by fmoralesc** [comment]
I would be interested in knowing if vim8 could handle something like that. With channels, it can communicate with remote processes already, but I don't know if there are any nice facilities for something like nvimfs.

**Comment by mcepl** [comment]
There is something here https://github.com/roxma/nvim-yarp

**Comment by mcepl** [comment]
BTW, on the theme of various jump I only now discovered oldie but goldie which I saved from all backups as https://github.com/mcepl/utl.vim (of course, pure vimL, works with very ancient vims).
